File: app/src/main/java/com/example/thechallen_ge/Dialog.java
package com.example.thechallen_ge;

import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatDialogFragment;

public class Dialog extends AppCompatDialogFragment {

    private DialogListener listener;
    AlertDialog.Builder builder;

    public interface DialogListener {
        void onOkayClicked();
    }

    @Override
    public android.app.Dialog onCreateDialog(Bundle savedInstanceState) {
        builder = new AlertDialog.Builder(getActivity());
        builder.setPositiveButton("Ok", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                listener.onOkayClicked();
            }
        });

        if (getArguments() != null) {
            String title = getArguments().getString("title");
            String message = getArguments().getString("message");

            builder.setTitle(title);
            builder.setMessage(message);
        }

        return builder.create();
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        try {
            listener = (DialogListener) context;
        } catch (ClassCastException e) {

        }
    }
}

File: app/src/main/java/com/example/thechallen_ge/Dialog.kt
package com.example.thechallen_ge

import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatDialogFragment

class Dialog : AppCompatDialogFragment() {
// Test test
    //Test test
     private lateinit var listener: DialogListener
     private lateinit var builder: AlertDialog.Builder

    interface DialogListener {
        fun onOkayClicked()
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): android.app.Dialog {
        builder = AlertDialog.Builder(activity!!)
        builder.setPositiveButton("Ok") { dialog, which -> listener.onOkayClicked() }

        if (arguments != null) {
            val title = arguments!!.getString("title")
            val message = arguments!!.getString("message")

            builder.setTitle(title)
            builder.setMessage(message)
        }

        return builder.create()
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        try {
            listener = context as DialogListener
        } catch (e: ClassCastException) {

        }

    }
}